pkg/cvo/cvo.go

```diff
@@ -160,6 +163,25 @@ func New(
 
 		queue:                 workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "clusterversion"),
 		availableUpdatesQueue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "availableupdates"),
+		proxyExists:           false,
+	}
+
+	discover := kubeClient.Discovery()
+	apiResourcesList, err := discover.ServerResourcesForGroupVersion("config.openshift.io/v1")
```

Comment:
The discovery is not useful.. this will always be true for CVO start...

the proxy CRD is created by an top-level operator cluster-config-operator... that is created by cluster-version-operator.

Reply:
If that is the case how would we check if Proxy kind exists?  It seems like any check would fail based on the above description.

Reply:
when you go to `GET` proxy object and you get not found, you don't use it..

Reply:
if the proxy api exists and the resource exists, you will be able to GET successfully and then we should start using the Proxy codepath.